src/client/InteractionClient.js

```diff
@@ -0,0 +1,232 @@
+'use strict';
+
+const BaseClient = require('./BaseClient');
+const ApplicationCommand = require('../structures/ApplicationCommand');
+const CommandInteraction = require('../structures/CommandInteraction');
+const { Events, ApplicationCommandOptionType, InteractionType, InteractionResponseType } = require('../util/Constants');
+
+let sodium;
+
+function transformCommand(command) {
+  return {
+    name: command.name,
+    description: command.description,
+    options: command.options?.map(function m(o) {
+      return {
+        type: ApplicationCommandOptionType[o.type],
+        name: o.name,
+        description: o.description,
+        default: o.default,
+        required: o.required,
+        choices: o.choices,
+        options: o.options?.map(m),
+      };
+    }),
+  };
+}
+
+/**
+ * Interaction client is used for interactions.
+ *
+ * @example
+ * const client = new InteractionClient({
+ *   token: ABC,
+ *   publicKey: XYZ,
+ * });
+ *
+ * client.on('interactionCreate', (interaction) => {
+ *   // automatically handles long responses
+ *   if (will take a long time) {
+ *     doSomethingLong.then((d) => {
+ *       interaction.reply({
+ *         content: 'wow that took long',
+ *       });
+ *     });
+ *   } else {
+ *     interaction.reply('hi!');
+ *   }
+ * });
+ * ```
+ */
+class InteractionClient extends BaseClient {
+  /**
+   * @param {Options} options Options for the client.
+   * @param {undefined} client For internal use.
+   */
+  constructor(options, client) {
+    super(options);
+
+    Object.defineProperty(this, 'token', {
+      value: options.token,
+      writable: true,
+    });
+
+    if (client) {
+      this.client = client;
+    } else {
+      this.client = this;
+      this.interactionClient = this;
+
+      Object.defineProperty(this, 'applicationID', {
+        value: options.clientID,
+        writable: true,
+      });
+
+      Object.defineProperty(this, 'publicKey', {
+        value: options.publicKey ? Buffer.from(options.publicKey, 'hex') : undefined,
+        writable: true,
+      });
+    }
+  }
+
+  /**
+   * Fetch registered slash commands.
+   * @param {Snowflake} [guildID] Optional guild ID.
+   * @returns {ApplicationCommand[]}
+   */
+  async fetchCommands(guildID) {
+    let path = this.client.api.applications(this.applicationID);
+    if (guildID) {
+      path = path.guilds(guildID);
+    }
+    const commands = await path.commands.get();
+    return commands.map(c => new ApplicationCommand(this, c, guildID));
+  }
+
+  /**
+   * Set all the commands for the application or guild.
+   * @param {Object[]} commands The command descriptor.
+   * @param {Snowflake} [guildID] Optional guild ID.
+   * @returns {ApplicationCommand[]} The commands.
+   */
+  async setCommands(commands, guildID) {
```

Comment:
This is better described as createCommands IMO, setCommands to me sounds like it would update or add commands, while this can only be used to add commands. However, if this was changed to use the (still  undocumented) PUT method, the name would be more accurate.